test/plugin/watch_test.rb

```diff
@@ -36,24 +36,21 @@ def thread_current_running?
     Thread.current[:namespace_watch_retry_count] = 0
 
     @client = OpenStruct.new
-    def @client.resourceVersion
-      '12345'
-    end
     def @client.watch_pods(options = {})
       []
     end
     def @client.watch_namespaces(options = {})
       []
     end
     def @client.get_namespaces(options = {})
-      self
+      {items: [], metadata: {resourceVersion: '12345'}}
     end
     def @client.get_pods(options = {})
-      self
+      {items: [], metadata: {resourceVersion: '12345'}}
     end
 
     @exception_raised = OpenStruct.new
-    def @exception_raised.each
+    def @exception_raised.[](_)
```

Comment:
This one is a bit strange.  It's used in 2 tests like this:
```
        @client.stub :watch_namespaces, [[@created, @exception_raised]] do
```
Is the array used to simulate the watch object .each first yielding a value, then raising an exception? But instead of the watch object raising, it yields an "explosive" value that raises when it gets accessed?

Reply:
made this a bit more obvious ... it was meant to cause an exception when being used, since "correct" notices do not blow up, but any undefined behavior is rescued